packages/firebase_database/lib/src/firebase_database.dart

```diff
@@ -15,11 +15,24 @@ class FirebaseDatabase {
   static final Map<int, StreamController<Event>> _observers =
       <int, StreamController<Event>>{};
 
+  static final Map<int, TransactionHandler> _transactions =
+      <int, TransactionHandler>{};
+
   FirebaseDatabase._() {
     _channel.setMethodCallHandler((MethodCall call) {
       if (call.method == 'Event') {
         final Event event = new Event._(call.arguments);
         _observers[call.arguments['handle']].add(event);
+      } else if (call.method == 'DoTransaction') {
+        final MutableData mutableData =
+            new MutableData._(call.arguments['snapshot']);
+        return _transactions[call.arguments['transactionKey']](mutableData)
+            .then<Map<String, dynamic>>((MutableData updated) {
+          return <String, dynamic>{'value': updated.value};
+        });
+      } else {
+        throw new MissingPluginException(
+            '${call.method} method not implemented on the Dart side.');
```

Comment:
Put a trailing comma after the string and linebreak; see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation

The easy way to remember this is if you see the formatter do a double-indent, try adding more trailing commas. Trailing commas are great!

Reply:
Done!

Reply:
Done!